src/components/Reports/DonationsReport/DonationsReportTable.tsx

```diff
@@ -1,10 +1,242 @@
+          donation.foreignAmount;
+        foreignTotals[donation.foreignCurrency].convertedTotal +=
+          donation.convertedAmount;
+      } else {
+        foreignTotals[donation.foreignCurrency] = {
+          convertedTotal: donation.convertedAmount,
+          foreignTotal: donation.foreignAmount,
+        };
+      }
+    });
+
+    return foreignTotals;
+  };
+
+  const foreignTotals = calculateForeignTotals();
+
+  return (
+    <>
+      <Box style={{ display: 'flex', margin: 8 }}>
+        <Typography variant="h6">{title}</Typography>
+        <Button
+          style={{ marginLeft: 'auto' }}
+          variant="contained"
+          startIcon={<ChevronLeftIcon />}
+          size="small"
+          onClick={() => setPrevMonth()}
+        >
+          Previous Month
+        </Button>
+        <Button
+          variant="contained"
+          endIcon={<ChevronRightIcon />}
+          size="small"
+          onClick={() => setNextMonth()}
+          disabled={hasNext}
+        >
+          Next Month
+        </Button>
+      </Box>
+      <Divider style={{ margin: 12 }} variant="middle"></Divider>
+      {!isEmpty ? (
+        <Data>
+          <DataGrid
+            rows={data}
+            columns={columns}
+            autoPageSize
+            autoHeight
+            disableSelectionOnClick
+            hideFooter
+          />
+          <Table>
+            {Object.entries(foreignTotals).map(([currency, total]) => (
+              <TableRow key={currency}>
+                <TableCell style={{ width: 375 }}>
+                  <Typography style={{ float: 'right', fontWeight: 'bold' }}>
+                    {t('Total ' + currency + ' Donations:')}
+                  </Typography>
+                </TableCell>
+                <TableCell style={{ width: 160 }}>
+                  <Typography style={{ float: 'left', fontWeight: 'bold' }}>
+                    {' ' + total.convertedTotal + ' ' + accountCurrency}
+                  </Typography>
+                </TableCell>
+                <TableCell style={{}}>
+                  <Typography style={{ float: 'left', fontWeight: 'bold' }}>
+                    {' ' + total.foreignTotal + ' ' + currency}
+                  </Typography>
+                </TableCell>
+              </TableRow>
+            ))}
+            <TableRow>
+              <TableCell>
+                <Typography style={{ float: 'right', fontWeight: 'bold' }}>
+                  {t('Total Donations: ')}
+                </Typography>
+              </TableCell>
+              <TableCell>
+                <Typography style={{ float: 'left', fontWeight: 'bold' }}>
+                  {totalDonations}
+                </Typography>
+              </TableCell>
+
+              <TableCell />
+            </TableRow>
+          </Table>
+        </Data>
+      ) : (
+        <EmptyDonationsTable
+          title={'No donations received in ' + time.monthLong + ' ' + time.year}
```

Comment:
Needs translation and interpolation
```suggestion
          title={t('No donations received in {{month}} {{year}}', { month: time.monthLong, year: time.year })}
```